iam-login-service/src/main/webapp/WEB-INF/views/iam/login.jsp

```diff
@@ -72,62 +72,129 @@
             <h3>Welcome to <strong>${iamOrganisationName}</strong>
       </h3>
         </div>
+
+        <c:if
+            test="${
+                (
+                    loginPageConfiguration.localAuthenticationVisible
+                    or param.sll != null
+                ) && loginPageConfiguration.defaultLoginPageLayout
+            }"
+        >
+            <jsp:include page="login-form.jsp" />
+        </c:if>
         
-        <c:if test="${loginPageConfiguration.localAuthenticationVisible or param.sll != null}">
-          <jsp:include page="login-form.jsp" />
+        <c:if
+            test="${
+                !loginPageConfiguration.defaultLoginPageLayout
+                || (
+                    !loginPageConfiguration.localAuthenticationVisible
+                    && (param.sll == null or !loginPageConfiguration.defaultLoginPageLayout)
+                )
+            }"
+        >
+            <c:set value="container-spacer" var="containerSpacer"></c:set>
         </c:if>
         
-        <div id="login-external-authn">
+        <div id="login-external-authn" class="${containerSpacer}">
             <c:if
-        test="${loginPageConfiguration.oidcEnabled or loginPageConfiguration.samlEnabled or IAM_X509_CAN_LOGIN}">
-              <div class="ext-login-preamble text-muted">
+                test="${loginPageConfiguration.oidcEnabled or loginPageConfiguration.samlEnabled or IAM_X509_CAN_LOGIN}"
+            >
+                <div class="ext-login-preamble text-muted">
+                    <c:choose>
+                        <c:when
+                            test="${
+                                (
```

Comment:
Isn't this equivalent to `loginPageConfiguration.defaultLoginPageLayout && (loginPageConfiguration.localAuthenticationVisible || param.sll != null)`

Reply:
Simplified the logic. Done